### ISSUE TITLE
Renew local environment Setup

### ISSUE THREAD

**Issue by amrap030** [taia-x/taia-x#105]
Since we don't have a backend component anymore, the `docker-compose` file doesn't need a `db` and `server` component anymore. Furthermore, the `Dockerfile` for the frontend has to be rebuilt. We may also think about completely removing the `backend` directory from our repository.

**Comment by eliodemollari** [comment]
https://github.com/taia-x/taia-x/blob/a3a178a712b157736f7066958ffd200be75eedba/frontend/Dockerfile#L7

Hey Kevin, would you consider changing it to `yarn install`, it seems more stable to me, and `npm install` was causing some errors?

**Comment by eliodemollari** [comment]
```
#9 223.1 [4/4] Building fresh packages...
#9 224.7 error /opt/taia-x/frontend/node_modules/deasync: Command failed.
#9 224.7 Exit code: 1
#9 224.7 Command: node ./build.js
#9 224.7 Arguments:
#9 224.7 Directory: /opt/taia-x/frontend/node_modules/deasync
#9 224.7 Output:
#9 224.7 gyp info it worked if it ends with ok
#9 224.7 gyp info using node-gyp@8.3.0
#9 224.7 gyp info using node@16.13.1 | linux | arm64
#9 224.7 gyp ERR! find Python
#9 224.7 gyp ERR! find Python Python is not set from command line or npm configuration
#9 224.7 gyp ERR! find Python Python is not set from environment variable PYTHON
#9 224.7 gyp ERR! find Python checking if "python3" can be used
#9 224.7 gyp ERR! find Python - "python3" is not in PATH or produced an error
#9 224.7 gyp ERR! find Python checking if "python" can be used
#9 224.7 gyp ERR! find Python - "python" is not in PATH or produced an error
#9 224.7 gyp ERR! find Python
#9 224.7 gyp ERR! find Python **********************************************************
#9 224.7 gyp ERR! find Python You need to install the latest version of Python.
#9 224.7 gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
#9 224.7 gyp ERR! find Python you can try one of the following options:
#9 224.7 gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
#9 224.7 gyp ERR! find Python (accepted by both node-gyp and npm)
#9 224.7 gyp ERR! find Python - Set the environment variable PYTHON
#9 224.7 gyp ERR! find Python - Set the npm configuration variable python:
#9 224.7 gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
#9 224.7 gyp ERR! find Python For more information consult the documentation at:
#9 224.7 gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
#9 224.7 gyp ERR! find Python **********************************************************
#9 224.7 gyp ERR! find Python
#9 224.7 gyp ERR! configure error
#9 224.7 gyp ERR! stack Error: Could not find any Python installation to use
#9 224.7 gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:330:47)
#9 224.7 gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:159:21)
#9 224.7 gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:202:16)
#9 224.7 gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:294:16)
#9 224.7 gyp ERR! stack     at exithandler (node:child_process:404:5)
#9 224.7 gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:416:5)
#9 224.7 gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
#9 224.7 gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
#9 224.7 gyp ERR! stack     at onErrorNT (node:internal/child_process:477:16)
#9 224.7 gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:83:21)
#9 224.7 gyp ERR! System Linux 5.10.76-linuxkit
#9 224.7 gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
#9 224.7 gyp ERR! cwd /opt/taia-x/frontend/node_modules/deasync
#9 224.7 gyp ERR! node -v v16.13.1
#9 224.7 gyp ERR! node-gyp -v v8.3.0
#9 224.7 gyp ERR! not ok
#9 224.7 Build failed
```

I'm also getting this error, which from what I could find on google, looks like node-gyp is failing because it can't find Python. Here is the link to the [issue](https://github.com/felixrieseberg/windows-build-tools/issues/56).  
I will install Python global, but the purpose of docker is that anyone can run it locally just by executing `docker-compose up`, so I don't think that this problem should occur at all. Maybe we have to look a bit more into this, what do you think?

**Comment by amrap030** [comment]
@eliodemollari I don't know, for me everything runs just fine via `docker compose up` or `docker compose up frontend`.

**Comment by amrap030** [comment]
@eliodemollari can you pull the latest version of the branch and try again? Beforehand I would suggest to delete the old container via `docker rm <container id>` and delete the image via `docker rmi <image id>`.

**Comment by eliodemollari** [comment]
@amrap030 Unfortunately it's not working, even though I deleted first all `images` & `containers`. It looks like a conflict between two dependencies:

```
While resolving: pinia@2.0.9
Found: typescript@4.1.6
```

Do you have any idea, because from what I read on google it occurs when using vue.js

**Comment by amrap030** [comment]
@eliodemollari I have no idea. We can look at it via Zoom if you want. Just hit me up on slack, to find a suitable timeslot. 

What happens when you run it without docker and only run `npm run serve` in the frontend folder?

**Comment by eliodemollari** [comment]
@amrap030 I changed the version of the two dependencies and now it seems to work

**Comment by repype** [comment]
@eliodemollari @amrap030 I have integrated a [dotenv file](https://github.com/taia-x/taia-x/blob/105-renew-local-setup/frontend/.env.template) for the frontend, so that we no longer track the contract address (which is different for everyone in the sandbox - **please keep in mind that you need to create the .env file**). If you now want to originate the contract in the testnet and use it in the frontend, you have to manually adjust the url and contract address inhere after deployment with `yarn deploy:testnet`. If you have ideas to solve this more elegantly (e.g. with two different modes - dev and staging - with which you can start the docker-compose services) I'm open to it, but I couldn't think of a better solution based on our use case (Against the two modes speaks for example that the staging mode is actually also local, because the frontend is not deployed anywhere).

**Comment by amrap030** [comment]
Merged via #108

**Comment by eliodemollari** [comment]
I was still having some troubles with the` deploy.js `. So I will reopen it and work today on the error.
